engines/python/setup/djl_python/rolling_batch/rolling_batch_service.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
+#
+# Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file
+# except in compliance with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS"
+# BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+from djl_python import Output, Input
+from djl_python.properties_manager.hf_properties import HuggingFaceProperties
+from djl_python.utils import read_model_config, get_tokenizer, parse_input_with_formatter, InputFormatConfigs
+from djl_python.rolling_batch.rolling_batch import get_content_type_from_output_formatter
+
+LMI_DIST_ADV_MODEL = {
+    "RWForCausalLM",
+    "GPTNeoXForCausalLM",
+    "T5ForConditionalGeneration",
+    "LlamaForCausalLM",
+    "FalconForCausalLM",
+    "MPTForCausalLM",
+    "GPTBigCodeForCausalLM",
+}
+
+
+def get_rolling_batch_class_from_str(rolling_batch_type: str, is_mpi: bool,
```

Comment:
does TRTLLM also falls into this category?

Reply:
This PR does not cover it. But it should be easy to re-use this code for trtllm as well.